### PR TITLE
Add Docs

### PR DESCRIPTION
**Pull request by andrewtimberlake**
I’ve added docs, doc tests.
Made some refactorings as I went.

### CODE REVIEW

File: mix.exs [hunk 1]
```diff
@@ -45,9 +45,9 @@ defmodule Money.Mixfile do
 
   defp package do
     [
-     maintainers: ["Giulio De Donato"],
-     contributors: ["Giulio De Donato"],
+     maintainers: ["Giulio De Donato", "Andrew Timberlake"],
+     contributors: ["Giulio De Donato", "Andrew Timberlake"],
```
Comment (liuggio) [inline mix.exs]: :+1:

File: lib/money.ex [hunk 2]
```diff
@@ -240,99 +269,168 @@ defmodule Money do
...
+  """
+  def multiply(%Money{amount: a, currency: cur}, %Money{amount: b, currency: cur}),
+    do: Money.new(a * b, cur)
+  def multiply(%Money{amount: amount, currency: cur}, multiplier) when is_integer(multiplier),
+    do: Money.new(amount * multiplier, cur)
+  def multiply(a, b), do: fail_currencies_must_be_equal(a, b)
+
+  @spec divide(t, t | integer) :: t
+  @doc ~S"""
+  Divide one `Money` from another or a `Money` with an integer
```
Comment (liuggio) [inline lib/money.ex]: typo: divides?

**Comment by liuggio** [comment]
impressive!!

**Comment by andrewtimberlake** [comment]
Thanks. Rebased against master and added docs for `Money.Currency`

**Pull request merged**